core/silkworm/consensus/engine.hpp

```diff
@@ -28,47 +28,66 @@
 
 namespace silkworm::consensus {
 
-class ConsensusEngine {
+class IConsensusEngine {
   public:
+    explicit IConsensusEngine(const ChainConfig& chain_config) : chain_config_{chain_config} {};
+    virtual ~IConsensusEngine() = default;
+
     //! \brief Performs validation of block header & body that can be done prior to sender recovery and execution.
     //! \brief See [YP] Sections 4.3.2 "Holistic Validity", 4.3.4 "Block Header Validity", and 11.1 "Ommer Validation".
     //! \param [in] block: block to pre-validate.
     //! \param [in] state: current state.
-    //! \param [in] config: current chain config.
     //! \note Shouldn't be used for genesis block.
-    virtual ValidationResult pre_validate_block(const Block& block, State& state, const ChainConfig& config) = 0;
+    virtual ValidationResult pre_validate_block(const Block& block, State& state) = 0;
 
     //! \brief See [YP] Section 4.3.4 "Block Header Validity".
     //! \param [in] header: header to validate.
     //! \param [in] state: current state.
-    //! \param [in] config: current chain config.
     //! \note Shouldn't be used for genesis block.
-    virtual ValidationResult validate_block_header(const BlockHeader& header, State& state,
-                                                   const ChainConfig& config) = 0;
+    virtual ValidationResult validate_block_header(const BlockHeader& header, State& state) = 0;
 
-    //! \brief See [YP] Section 11.3 "Reward Application".
+    //! \brief Validates the seal of the header
+    virtual ValidationResult validate_seal(const BlockHeader& header) {
+        (void)header;
+        return ValidationResult::kOk;
+    }
+
+    //! \brief Finalizes consensus tasks by applying changes in the state of accounts or of the consensus itself
```

Comment:
I'd say "Finalizes block execution" instead of "consensus tasks"

Reply:
Resolved by 6526c60